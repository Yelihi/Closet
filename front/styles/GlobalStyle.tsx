import { createGlobalStyle } from 'styled-components';
import normalize from 'styled-normalize';

export interface GlobalProps {
  isPhoneMenuClick: boolean;
}

const GlobalStyle = createGlobalStyle`
  ${normalize}

    /* font */

  @font-face {
    font-family: CastleBegaleTTF;
    src: url('/fonts/CastleBegaleTTF.ttf');
  }

  html {
    box-sizing: border-box;
    font-size: 62.5%;
    min-width: 320px;
    height: 100%;
  }

  body {
    height: 100%;
    overflow: ${(props: GlobalProps) => (props.isPhoneMenuClick ? 'hidden' : 'auto')};
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
    margin: 0;
  }

  * { font-family: ${({ theme }) => theme.font.Logo}}

  a {  cursor: pointer; text-decoration: none}

  a { text-decoration: none; outline: none}
  a:hover, a:active {text-decoration: none}

  input,
  input:active,
  input:focus{
    border: none;
    outline: none;
    -webkit-appearance:none;
    -moz-appearance: none;
    -o-appearance:none;
    appearance: none;
    font-family: ${({ theme }) => theme.font.Efont};
  }

  button {
    border: none;
    cursor: pointer;
    font-family: ${({ theme }) => theme.font.Efont};
  }

  ul,
  li,
  li:active,
  li:hover{
    list-style: none;
    padding: 0;
    margin: 0;

  }



  
`;

export default GlobalStyle;
