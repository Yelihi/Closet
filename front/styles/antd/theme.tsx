export const defaultTheme = {
  token: {
    fontSize: 12,
  },
};
