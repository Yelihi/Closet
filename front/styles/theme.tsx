import { DefaultTheme, CSSProp, css } from 'styled-components';

const font = {
  Kfont: `'Noto Sans KR', sans-serif`,
  Efont: `'Roboto', sans-serif`,
  Efont2: `'CastleBegaleTTF', sans-serif`,
  Logo: `'Julius Sans One', sans-serif`,
};

const fontWeight = {
  Thin: 100,
  Light: 300,
  Regular: 400,
  Medium: 500,
  Bold: 700,
};

const colors = {
  black: '#000000',
  white: '#FFFFFF',
  lightGrey: '#B0B0B0',
  middleGrey: '#717171',
  deepGrey: '#222222',
  hoverGrey: '#dbdcdc',
  mainGrey: '#EDECEC',
  symbol: '#205281',
  brown: '#363740',
  red: '#EE4B2B',
  success: '#68C260',
  warnning: '#F4A100',
  milk: '#f7f6ee',
};

export type FontTypes = typeof font;
export type FontWeight = typeof fontWeight;
export type Colors = typeof colors;

const theme: DefaultTheme = {
  breakPoint: '796px',
  font,
  fontWeight,
  colors,
};

export default theme;
