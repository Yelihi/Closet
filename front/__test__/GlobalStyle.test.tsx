import { render } from '@testing-library/react';

import GlobalStyle from '../styles/GlobalStyle';
import theme from '../styles/theme';

// 여긴 다시한번 체크를 해보자.. 스냅샷이 되는게 이상한데..
describe('Global Style', () => {
  it('render correctly', () => {
    render(<GlobalStyle isPhoneMenuClick={false} theme={theme} />);

    expect(document.head).toMatchSnapshot();
  });
});
