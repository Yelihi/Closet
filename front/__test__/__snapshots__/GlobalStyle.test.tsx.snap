// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Style render correctly 1`] = `<head />`;
